iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java

```diff
@@ -91,6 +97,35 @@ public void start() throws Exception {
         }
     };
 
+    public Set<String> pruneUnwantedImages() {
+        List<Image> images = docker.getImages();
+        LoggingService.logDebug(MODULE_NAME, "Total number of images already downloaded in the machine : " + images.size());
+        Set<String> imageIDsToBePruned = new HashSet<>();
+        List<Microservice> microservices = microserviceManager.getLatestMicroservices();
```

Comment:
How about non-iofog containers? Their images should be kept as well.

Reply:
Added check for non iofog containers.